src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,122 @@
+import {
+  memo,
+  useRef,
+  useState,
+  useEffect,
+} from 'react';
+import classNames from 'classnames';
+
+import { Action, Todo } from '../../types';
+import { useTodos } from '../../hooks';
+
+type Props = {
+  item: Todo;
+};
+
+export const TodoItem: React.FC<Props> = memo(({ item }) => {
+  const { dispatch } = useTodos();
+  const [isEditing, setIsEditing] = useState(false);
+  const [editedTitle, setEditedTitle] = useState(item.title);
+  const editInputRef = useRef<HTMLInputElement | null>(null);
+
+  const { id, title, completed } = item;
+
+  useEffect(() => {
+    if (isEditing && editInputRef.current) {
+      editInputRef.current.focus();
+    }
+  }, [isEditing]);
+
+  const submitEdit = () => {
+    if (!isEditing) {
+      return;
+    }
+
+    const newTitle = editedTitle.trim();
+
+    if (!newTitle) {
+      dispatch({
+        type: Action.Remove,
+        payload: id,
+      });
+    } else if (newTitle !== title) {
+      dispatch({
+        type: Action.Edit,
+        payload: { id, title: newTitle },
+      });
+    }
```

Comment:
```suggestion
    if (!newTitle) {
      dispatch({
        type: Action.Remove,
        payload: id,
      });
    } 
    if (newTitle !== title) {
      dispatch({
        type: Action.Edit,
        payload: { id, title: newTitle },
      });
    }
```

Reply:
It was like that at the beginning, but an empty title is also not equal to the current title (`'' !== 'something'`), so both dispatches will be called. That's why I need that `else if` there.